borg/repository.py

```diff
@@ -83,6 +85,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is not None:
             self.rollback()
+            if hasattr(exc_val, 'errno'):
+                if exc_val.errno in [errno.EIO, errno.ENXIO]:
+                    return
         self.close()
```

Comment:
But for input-caused, neither `errno.EIO` nor `errno.ENXIO` occurs; rather `errno.ENOENT` occurs.

errno.ENOENT  == No such file or directory
errno.EIO == I/O error
errno.ENXIO == No such device or address